harvester_e2e_tests/integrations/test_4_vm_backup_restore.py

```diff
@@ -626,6 +626,210 @@ def test_restore_replace_with_vm_shutdown_command(
+        pytest.param("NFS", marks=pytest.mark.NFS)
+    ],
+    indirect=True)
+class TestBackupRestoreWithSnapshot:
+
+    @pytest.mark.dependency()
+    def test_connection(self, api_client, backup_config, config_backup_target):
+        code, data = api_client.settings.backup_target_test_connection()
+        assert 200 == code, f'Failed to test backup target connection: {data}'
+
+    @pytest.mark.dependency(depends=["TestBackupRestoreWithSnapshot::test_connection"], param=True)
+    def tests_backup_vm(self, api_client, wait_timeout, backup_config, base_vm_with_data):
+        unique_vm_name = base_vm_with_data['name']
+
+        # Create backup with the name as VM's name
+        code, data = api_client.vms.backup(unique_vm_name, unique_vm_name)
+        assert 204 == code, (code, data)
+        # Check backup is ready
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, backup = api_client.backups.get(unique_vm_name)
+            if 200 == code and backup.get('status', {}).get('readyToUse'):
+                break
+            sleep(3)
+        else:
+            raise AssertionError(
+                f'Timed-out waiting for the backup \'{unique_vm_name}\' to be ready.'
+            )
+
+    @pytest.mark.dependency(depends=["TestBackupRestoreWithSnapshot::tests_backup_vm"], param=True)
+    def test_with_snapshot_restore_with_new_vm(
+        self, api_client, vm_shell_from_host, vm_checker, ssh_keypair, wait_timeout,
+        backup_config, base_vm_with_data
+    ):
+        unique_vm_name, backup_data = base_vm_with_data['name'], base_vm_with_data['data']
+        pub_key, pri_key = ssh_keypair
+
+        vm_snapshot_name = unique_vm_name + '-snapshot'
+        # take vm snapshot
+        code, data = api_client.vm_snapshots.create(unique_vm_name, vm_snapshot_name)
+        assert 201 == code
+
+        deadline = datetime.now() + timedelta(seconds=wait_timeout)
+        while deadline > datetime.now():
+            code, data = api_client.vm_snapshots.get(vm_snapshot_name)
+            if data.get("status", {}).get("readyToUse"):
+                break
+            print(f"waiting for {vm_snapshot_name} to be ready")
+            sleep(3)
+        else:
+            raise AssertionError(f"timed out waiting for {vm_snapshot_name} to be ready")
+
+        assert 200 == code
+        assert data.get("status", {}).get("readyToUse") is True
+
+        # mess up the existing data
+        with vm_shell_from_host(
+            base_vm_with_data['host_ip'], base_vm_with_data['vm_ip'],
+            base_vm_with_data['ssh_user'], pkey=pri_key
+        ) as sh:
+            out, err = sh.exec_command(f"echo {pub_key!r} > {base_vm_with_data['data']['path']}")
+            assert not err, (out, err)
+            sh.exec_command('sync')
+
+        # Restore VM into new
+        restored_vm_name = f"{backup_config[0].lower()}-restore-{unique_vm_name}"
```

Comment:
Same, consider use an unique name to avoid previous teardown fail (previous `restored_vm` still there).
![image](https://github.com/user-attachments/assets/7f68d9ec-e469-4ec5-a2b7-4f3706b4e4cc)

Reply:
Thanks for the suggestion. 
I set the restored_vm_name to use a more precise name to distinguish with other test class.